types/validation.go

```diff
@@ -67,7 +67,12 @@ func (b *Block) ValidateWithState(state *State) error {
 		return NewErrTimeFraud(b, currentTime)
 	}
 
-	if b.Header.Version.App != state.Version.Consensus.App ||
+	appVersion := state.Version.Consensus.App
+	if b.Header.Height < state.VersionStartHeight {
+		appVersion--
```

Comment:
if this change to support historic sync, how u know the expected revision is -1 and not more?

Reply:
this needs to be changes once we add historic snyc. this is a temporary hack to make it work by now assuming there is a single forkl

Reply:
pls add fixme

Reply:
done